### PR TITLE
Initial support for Heroku review apps

### PR DESCRIPTION
**Pull request by armenzg**

### CODE REVIEW

File: app.json [hunk 1]
```diff
@@ -13,87 +24,29 @@
...
-      "required": true
-    },
-    "SITE_URL": {
-      "required": true
+    "PULSE_TASKS_QUEUE_NAME": {
+      "generator": "secret"
     },
     "TREEHERDER_DJANGO_SECRET_KEY": {
-      "required": true
+      "value": "secret-key-of-at-least-50-characters-to-pass-check-deploy"
```
Comment (pmac) [inline app.json]: No `"generator": "secret"` here? Not long enough?

**Comment by armenzg** [comment]
You can see the Review app deployed in [here](https://treeherder-heroku-revie-kgzmk8.herokuapp.com/#/jobs?repo=autoland).

I have adjusted the workers so we only process pushes and tasks.
The dynos can always be adjusted *after* the Review app is first created with the formation specified `app.json`.

File: app.json (hunk 1 above)
Comment (armenzg) [reply app.json]: Addressed! 64 chars so we're good :)

**Comment by sarah-clements** [comment]
Sorry Armen, I didn't get a chance to look at this. I've tagged @camd for review so maybe he can look at it Monday.

File: app.json [hunk 2]
```diff
@@ -13,76 +13,13 @@
   ],
   "env": {
     "BROKER_URL": {
-      "required": true
-    },
```
Comment (camd) [inline app.json]: Why are we removing these requirements?  I'm not saying it's wrong to do so.  I just don't know this area well enough to know why these are bad to have.  If I understand, these will help us ensure we set these required values on a new instance?

File: app.json [hunk 3]
```diff
@@ -39,6 +39,9 @@
     "PULSE_TASKS_QUEUE_NAME": {
       "generator": "secret"
     },
+    "PROJECTS_TO_INGEST": {
+      "value": "autoland,android-components"
```
Comment (camd) [inline app.json]: It would be valuable to have ``try`` in this list.  I know it is larger, because it has a lot of pushes.  But for folks that are ``level 1`` commit access, it is the only place I can do retriggers and other similar actions.  So it reduces the usefulness of these apps.

File: app.json (hunk 2 above)
Comment (armenzg) [reply app.json]: The original app.json was generated automatically when I enabled the feature Heroku Review Apps. It's only now that I have spent time figurying out which ones we actually do need.

`"required" true` is the same as saying copy the env variable from the app we're basing this off.

In some cases, we don't need them because we can't configure the Heroku scheduler to run the command to update the bugs cache.

In other cases, the env variables get defined when the add-on is added.

In other words, we might be able to use some of these env variables in the future if we invest more but for now this is the minimum amount we need to get the ingestion pipeline working, the backend and the frontend.

@camd does this answer satisfy you?

File: app.json (hunk 3 above)
Comment (armenzg) [reply app.json]: @camd this value can be adjusted after an app is created directly on Heroku.

The more we ingest the more a Review App costs. I would rather increase the ingestion cost by default only if necessary. At the moment, it costs $30/month. The next tier for the DB+AMQP totals $130/month. We might have to but I would like to wait and see.

Comment (camd) [reply app.json]: Fair enough.  That sounds pragmatic.  :)

File: app.json (hunk 2 above)
Comment (camd) [reply app.json]: Yep, makes sense.  Thanks for explaining.  :)
